python/cudf/cudf/core/series.py

```diff
@@ -2435,9 +2437,18 @@ def corr(self, other, method="pearson", min_periods=None):
         if self.empty or other.empty:
             return cudf.utils.dtypes._get_nan_for_dtype(self.dtype)
 
-        lhs = self.nans_to_nulls().dropna()
-        rhs = other.nans_to_nulls().dropna()
-        lhs, rhs = _align_indices([lhs, rhs], how="inner")
+        if method == "pearson":
+            lhs = self.nans_to_nulls().dropna()
+            rhs = other.nans_to_nulls().dropna()
+            lhs, rhs = _align_indices([lhs, rhs], how="inner")
+        elif method == "spearman":
+            lhs = self.nans_to_nulls().dropna()
+            rhs = other.nans_to_nulls().dropna()
+            lhs, rhs = _align_indices([lhs, rhs], how="inner")
+            lhs = lhs.rank()
+            rhs = rhs.rank()
+        else:
+            raise ValueError("method must be either 'pearson', 'spearman'")
```

Comment:
Similarly to reduce duplication:

```suggestion
        lhs = self.nans_to_nulls().dropna()
        rhs = other.nans_to_nulls().dropna()
        lhs, rhs = _align_indices([lhs, rhs], how="inner")
        if method == "spearman":
            lhs = lhs.rank()
            rhs = rhs.rank()
```